### PR TITLE
[Docs] Replace old code new_function() with set_function()

### PR DESCRIPTION
**Pull request by jillnogold**
[ML-6271](https://iguazio.atlassian.net/browse/ML-6271)

[ML-6271]: https://iguazio.atlassian.net/browse/ML-6271?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/runtimes/serving-function.md [hunk 1]
```diff
@@ -31,9 +31,9 @@ project = mlrun.get_or_create_project(
 serving_function_image = "mlrun/mlrun"
 serving_model_class_name = "mlrun.frameworks.sklearn.SklearnModelServer"
 
-# Create a serving function
-serving_fn = mlrun.new_function(
-    "serving", project=project.name, kind="serving", image=serving_function_image
+# Create a serving function within a project
+serving_fn = mlrun.project.set_function(
```
Comment (TomerShor) [inline docs/runtimes/serving-function.md]:
```suggestion
serving_fn = project.set_function(
```

File: docs/runtimes/serving-function.md [hunk 2]
```diff
@@ -31,9 +31,9 @@ project = mlrun.get_or_create_project(
...
+    "serving", kind="serving", image=serving_function_image
```
Comment (TomerShor) [inline docs/runtimes/serving-function.md]:
```suggestion
    name="serving", kind="serving", image=serving_function_image
```

File: docs/serving/built-in-model-serving.md [hunk 3]
```diff
@@ -46,8 +46,8 @@ serving_function_image = "mlrun/mlrun"
 serving_model_class_name = "mlrun.frameworks.sklearn.SklearnModelServer"
 
 # Create a serving function
-serving_fn = mlrun.new_function(
-    "serving", project=project.name, kind="serving", image=serving_function_image
+serving_fn = mlrun.project.set_function(
+    "serving", kind="serving", image=serving_function_image
```
Comment (TomerShor) [inline docs/serving/built-in-model-serving.md]:
```suggestion
serving_fn = project.set_function(
    name="serving", kind="serving", image=serving_function_image
```

File: docs/runtimes/spark-operator.ipynb [hunk 4]
```diff
@@ -79,7 +79,9 @@
     "# command will use our spark code which needs to be located on our file system\n",
     "# the name param can have only non capital letters (k8s convention)\n",
     "read_csv_filepath = os.path.join(os.path.abspath(\".\"), \"spark_read_csv.py\")\n",
-    "sj = mlrun.new_function(kind=\"spark\", command=read_csv_filepath, name=\"sparkreadcsv\")\n",
+    "sj = mlrun.project.set_function(\n",
```
Comment (TomerShor) [inline docs/runtimes/spark-operator.ipynb]: Also here, pretty sure you need to have a project instance first and not do `mlrun.project`

**Review by TomerShor — CHANGES_REQUESTED** [review]

File: docs/runtimes/create-and-use-functions.ipynb [hunk 5]
```diff
@@ -67,7 +67,7 @@
     "- [**Import existing function**](#import-existing-function) &mdash; when your function already exists elsewhere and you just want to import it\n",
     "\n",
     "``` {admonition} Note\n",
-    "Using the `set_function` method of an MLRun project allows for each of these scenarios in a transparent way. Depending on the source passed in, the project registers the function using some lower level functions. For specific use cases, you also have access to the lower level functions {py:func}`~mlrun.run.new_function`, {py:func}`~mlrun.run.code_to_function`, and {py:func}`~mlrun.run.import_function`.\n",
+    "Using the `set_function` method of an MLRun project allows for each of these scenarios in a transparent way. Depending on the source passed in, the project registers the function using some lower level functions. For specific use cases, you also have access to the lower level functions {py:func}`~mlrun.run.code_to_function` and {py:func}`~mlrun.run.import_function`.\n",
```
Comment (alonmr) [inline docs/runtimes/create-and-use-functions.ipynb]: new_function can still be used in this context. no need to remove

**Review by alonmr — CHANGES_REQUESTED** [review]
As Tomer said. You can't use `mlrun.project.set_function` you need a project instance. For example:
```
project = mlrun.get_or_create_project(...)
project.set_function(...)
```

File: docs/runtimes/spark-operator.ipynb [hunk 6]
```diff
@@ -79,7 +79,10 @@
     "# command will use our spark code which needs to be located on our file system\n",
     "# the name param can have only non capital letters (k8s convention)\n",
     "read_csv_filepath = os.path.join(os.path.abspath(\".\"), \"spark_read_csv.py\")\n",
-    "sj = mlrun.new_function(kind=\"spark\", command=read_csv_filepath, name=\"sparkreadcsv\")\n",
+    "project = mlrun.get_or_create_project(spark - proj)\n",
+    "sj = mlrun.project.set_function(\n",
```
Comment (TomerShor) [inline docs/runtimes/spark-operator.ipynb]:
```suggestion
    "sj = project.set_function(\n",
```

File: docs/runtimes/create-and-use-functions.ipynb [hunk 7]
```diff
@@ -431,13 +431,11 @@
    "metadata": {},
    "source": [
     "```python\n",
-    "fn = mlrun.new_function(\"serving\", kind=\"serving\", image=\"mlrun/mlrun\")\n",
+    "fn = mlrun.project.set_function(name=\"serving\", kind=\"serving\", image=\"mlrun/mlrun\", with_repo=True)\n",
```
Comment (TomerShor) [inline docs/runtimes/create-and-use-functions.ipynb]:
```suggestion
    "project = mlrun.get_or_create_project(\"my-project\")\n",
    "fn = project.set_function(name=\"serving\", kind=\"serving\", image=\"mlrun/mlrun\", with_repo=True)\n",
```

File: docs/runtimes/spark-operator.ipynb [hunk 8]
```diff
@@ -79,7 +79,10 @@
     "# command will use our spark code which needs to be located on our file system\n",
     "# the name param can have only non capital letters (k8s convention)\n",
     "read_csv_filepath = os.path.join(os.path.abspath(\".\"), \"spark_read_csv.py\")\n",
-    "sj = mlrun.new_function(kind=\"spark\", command=read_csv_filepath, name=\"sparkreadcsv\")\n",
+    "project = mlrun.get_or_create_project(spark - proj)\n",
```
Comment (TomerShor) [inline docs/runtimes/spark-operator.ipynb]:
```suggestion
    "project = mlrun.get_or_create_project(\"spark-proj\")\n",
```

File: docs/serving/model-serving-get-started.ipynb [hunk 9]
```diff
@@ -639,7 +639,7 @@
     }
    ],
    "source": [
-    "fn_preprocess2 = mlrun.new_function(\"preprocess\", kind=\"serving\")\n",
+    "fn_preprocess2 = mlrun.project.set_function(\"preprocess\", kind=\"serving\")\n",
```
Comment (TomerShor) [inline docs/serving/model-serving-get-started.ipynb]:
```suggestion
    "project = mlrun.get_or_create_project(project_name)\n",
    "fn_preprocess2 = project.set_function(\"preprocess\", kind=\"serving\")\n",
```

File: docs/serving/model-serving-get-started.ipynb [hunk 10]
```diff
@@ -747,7 +747,7 @@
    "source": [
     "import mlrun\n",
     "\n",
-    "fn_preprocess2 = mlrun.new_function(\"preprocess\", kind=\"serving\")\n",
+    "fn_preprocess2 = mlrun.project.set_function(\"preprocess\", kind=\"serving\")\n",
```
Comment (TomerShor) [inline docs/serving/model-serving-get-started.ipynb]:
```suggestion
    "fn_preprocess2 =project.set_function(\"preprocess\", kind=\"serving\")\n",
```

File: docs/runtimes/create-and-use-functions.ipynb (hunk 7 above)
Comment (gtopper) [inline docs/runtimes/create-and-use-functions.ipynb:434]: Pretty sure here too, `mlrun.project.set_function` shouldn't be called as if it were a static method.

**Review by TomerShor — CHANGES_REQUESTED** [review]
Found some more wrong usages of `mlrun.project`.

**Review by alonmr — APPROVED** [review]

**Review by TomerShor — APPROVED** [review]

**Pull request merged**